javascript/net/grpc/web/grpcwebclientbase.js

```diff
@@ -216,10 +216,16 @@ GrpcWebClientBase.prototype.processHeaders_ = function(xhr) {
     var currentTime = (new Date()).getTime();
     var timeout = Math.round(deadline - currentTime);
     xhr.headers.remove('deadline');
-    if (timeout > 0) {
+    if (timeout === Infinity) {
+      // Polyfill for "Number.MAX_SAFE_INTEGER"
+      var maxSafeInteger = Math.pow(2, 53) - 1;
+      timeout = maxSafeInteger;
+    }
+    // isFinite discards NaN and -Infinity
+    if (isFinite(timeout) && timeout > 0) {
```

Comment:
Do we support IE 11 ? 
If `yes` we need to Polyfill `isFinite()`
If `no` we can use `isFinite()` and `Number.MAX_SAFE_INTEGER` (ie. no polyfill)
https://caniuse.com/#feat=es6-number

Reply:
Should i create a polyfill for isFinite, just in case? Not sure if IE 11 should be supported or not, but fixing the polyfill in this case should be as simple as:

```javascript
...
if (timeout !== NaN && timeout !== -Infinity && timeout > 0) { ... }
```

right?

Reply:
I feel like @rogchap is talking about `Number.isFinite`, while this is `window.isFinite` which was already part of ES3 (Section 15.1.2.5 of https://www-archive.mozilla.org/js/language/E262-3.pdf).
I don't think a polyfill is required.

If we add the polyfill, it's sufficient to check `timeout !== Infinity` and `timeout > 0`.
`timeout !== NaN` is always true, because, accoring to the IEEE standard, comparing any number to `NaN` is false (including `NaN`).
```javascript
if (timeout !== Infinity && timeout > 0) { ... }
```

Reply:
Thanks @Yannic for providing the info on `window.isFinite` vs. `Number.isFinite` and `NaN !== NaN`. I didn't know this so I think it was very helpful. Perhaps sticking with `isFinite()` and no polyfill is the way to go then?

Reply:
Or wait... Perhaps we don't need `isFinite()` at all then.

There is already a check for `Infinity` on line 219, `timeout > 0` removes `-Infinity` and as @Yannic says, comparing any number to `NaN` is always false so `timeout > 0` is false for `NaN`. Which I think means there are only positive numbers left, which should all be allowed. I believe that sounds like the way to go.

Reply:
You're right that `timeout > 0` is sufficient!. I must have missed the check for `Infinity` above.

That being said, I think we should set the timeout to `2147483647 // 2^31 - 1` to be on the save side. Servers might use 32-bit signed integers for the timeout which will overflow for `2^53 - 1` Also, pre-calculating constants and commenting them is, IMO, better than calculating them every single time they're needed (not that it makes a big difference in this case).

Otherwise LGTM.

Reply:
I added the above so now I think the code should be finished.